mbed_host_tests/__init__.py

```diff
@@ -501,6 +495,8 @@ def execute(self):
         except Exception, e:
             print str(e)
             self.print_result(self.RESULT_ERROR)
+        finally:
+            self.test_supervisor.teardown()
```

Comment:
It can raise self.test_supervisor used before assignment. I think there should be a try except finally block around 

```python
self.test_supervisor.setup()
result = self.test_supervisor.test(self)
```